### PR TITLE
[event format - PART 3] Structured Cloud Event and Abstract Marshaller

### PR DESCRIPTION
**Pull request by michlee1337**
Creates the StructuredCloudEvent class and abstract Marshaller class that the rest of event formatting is based on.

Changes unique to this branch (from event format - PART 2 onwards) are from commit hash c6a1fbb onwards.
- [ x ] Tests pass
- [ x ] Appropriate changes to README are included in PR

### CODE REVIEW

File: v1/format.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        StructuredCloudEvent(CloudEventFormat format, std::string serialized_cloud_event);
+        CloudEventFormat GetCloudEventFormat();
+        std::string GetSerializedCloudEvent();
+};
+
+class Marshaller {
+    // interface for stand-alone format marshallers
+    public:
+        virtual absl::StatusOr<StructuredCloudEvent> Serialize(io::cloudevents::v1::CloudEvent cloud_event) = 0;
+        virtual absl::StatusOr<StructuredCloudEvent> Deserialize(std::string serialized_cloud_event) = 0;
```
Comment (hrasadi) [inline v1/format.h:27]: Shouldn't this return a CloudEvent? Also, I think it should accept a StructuredCloudEvent.

File: v1/format.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    private:
+        CloudEventFormat format_;
+        std::string serialized_cloud_event_;
+    public:
+        StructuredCloudEvent(CloudEventFormat format, std::string serialized_cloud_event);
+        CloudEventFormat GetCloudEventFormat();
+        std::string GetSerializedCloudEvent();
+};
+
+class Marshaller {
```
Comment (hrasadi) [inline v1/format.h:23]: Consider moving Marshaller to a separate file.

File: third_party/base64/base64.cc [hunk 3]
```diff
@@ -0,0 +1,246 @@
+/*
```
[the remark this answers is not in the document]
Comment (michlee1337) [reply third_party/base64/base64.cc:1]: Done!

**Comment by michlee1337** [comment]
Refactoring for readability.

**Pull request closed without merging**